### PR TITLE
Issue #24299: Add History to the search engine menu

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt [hunk 1]
```diff
@@ -44,6 +44,10 @@ interface SearchController {
     fun handleSearchTermsTapped(searchTerms: String)
     fun handleSearchShortcutEngineSelected(searchEngine: SearchEngine)
     fun handleClickSearchEngineSettings()
+    /**
+     * Handle when when search icon is tapped.
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt:48]:
```suggestion
     * Handle when search icon is tapped.
```

File: app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt [hunk 2]
```diff
@@ -211,6 +215,12 @@ class SearchDialogController(
         navController.navigateSafe(R.id.searchDialogFragment, directions)
     }
 
+    override fun handleClickHistorySearch() {
+        clearToolbarFocus()
+        val directions = SearchDialogFragmentDirections.actionGlobalHistoryFragment(search = true)
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt:220]: I think we already talked about this before, but were we landing the 'hacky' solution while working towards a true solution or were there some new developments where the 'hacky' solution is the true solution?

Comment (rocketsroger) [reply app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt:220]: This hacky solution is the cleanest solution I can find at the moment.  Tried to add in A-C side or on Fenix side as a search engine but keep running into issues that require fixes that create more problems.

File: app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt (hunk 1 above)
Comment (rocketsroger) [reply app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt:48]: I'll update.  Good catch, thanks

File: app/src/main/res/navigation/nav_graph.xml [hunk 3]
```diff
@@ -270,6 +270,10 @@
         android:name="org.mozilla.fenix.library.history.HistoryFragment"
         android:label="@string/library_history"
         tools:layout="@layout/fragment_history">
+        <argument
+            android:name="search"
```
Comment (gabrielluong) [inline app/src/main/res/navigation/nav_graph.xml:274]: Maybe this should be `showSearch` or some variant?

**Review by gabrielluong — APPROVED** [review]

**Comment by rocketsroger** [comment]
I think I found a better solution for this.  I'll keep this PR with do-not-land until I confirm.  Thanks

**Comment by rocketsroger** [comment]
Found a better solution.  Closing this

**Pull request closed without merging**